signup/tool/src/java/org/sakaiproject/signup/tool/jsf/SignupMeetingsBean.java

```diff
@@ -591,8 +591,14 @@ private void markingRecurMeetings(List<SignupMeetingWrapper> smList) {
 	 *         current user.
 	 */
 	public boolean isAllowedToDelete() {
-		if (getSignupMeetings() == null)
+		if (sakaiFacade.isAllowedSite(sakaiFacade.getCurrentUserId(), SakaiFacade.SIGNUP_DELETE_SITE, sakaiFacade.getCurrentLocationId())) {

```

Comment:
I'd just drop this shortcut as it doesn't really fix the performance issue and without the context of it specifically being a performance hack makes the it harder to read.